tests/phpunit/api/v3/CaseRoleCreationTest.php

```diff
@@ -110,6 +110,50 @@ public function testCaseRolePostProcessingIsNotTriggeredForNonCaseRelatedRelatio
     $this->assertCount(0, $activity['values']);
   }
 
+  /**
+   * Test an existent relationship is not created again.
+   */
+  public function testExistentRelationshipIsNotCreatedAgain() {
+    $caseClient = ContactFabricator::fabricate();
+    $contact = ContactFabricator::fabricate();
+    $relationshipParams = $this->getRelationshipBaseParamsForCaseClient($caseClient);
+
+    $relationshipParams['contact_id_b'] = $contact['id'];
+    $relationshipCreated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipCreatedAgain = civicrm_api3('Relationship', 'create', $relationshipParams);
+
+    $this->assertEquals($relationshipCreated['id'], $relationshipCreatedAgain['id']);
+  }
+
+  /**
+   * Test an existent relationship is correctly updated.
+   */
+  public function testExistentRelationshipIsCorrectlyUpdated() {
+    $caseClient = ContactFabricator::fabricate();
+    $firstContact = ContactFabricator::fabricate();
+    $secondContact = ContactFabricator::fabricate();
+    $relationshipParams = $this->getRelationshipBaseParamsForCaseClient($caseClient);
+
+    // Create first relationship.
+    $relationshipParams['contact_id_b'] = $firstContact['id'];
+    $relationshipCreated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipCreatedId = array_shift($relationshipCreated['values'])['id'];
+
+    // Update relationship with second contact.
+    $relationshipParams['contact_id_b'] = $secondContact['id'];
+    $relationshipUpdated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipUpdated = array_shift($relationshipUpdated['values']);
+
+    // Refresh previous relationship.
+    $relationshipCreated = new CRM_Contact_DAO_Relationship();
+    $relationshipCreated->id = $relationshipCreatedId;
+    $relationshipCreated->find();
+
+    $this->assertNotEquals($relationshipCreated->is_active, $relationshipUpdated['id']);
```

Comment:
Is this comparison correct comparing the is_active field with the relationship Id?

Reply:
Ooops, no no, it is not. Fixed here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/735/commits/e84495b0ace1b1db421cead1d031991fc7df3f38